### ISSUE TITLE
Bug: Jupyter Notebook trim whitespaces not working in code cells

### ISSUE THREAD

**Issue by hegerdes** [microsoft/vscode#181905]
Does this issue occur when all extensions are disabled?: Yes
Note: All extensions disabled - except python and jupyter notebook 

The function to trim whitespaces does not work in jupyter notebook code cells

- VS Code Version:  1.78.0 (252e5463d60e63238250799aef7375787f68b4ee) - x64
- OS Version: Win 11 Pro

Steps to Reproduce:

1. Create a new Jupyter Notebook (.ipynb) file
2. Create a cell and put some code with tailing whitespaces and empty lines in it

Example:
<img width="706" alt="image" src="https://github.com/microsoft/vscode/assets/40325561/52340bf3-bc78-4a7e-ac31-1bfaf46c68a5">

**Comment by rebornix** [comment]
It seems I can't reproduce in latest VS Code

https://github.com/microsoft/vscode/assets/876920/21382526-601f-42a2-9323-5e8a965ad7ac

@hegerdes can you please check if you can reproduce with all extensions disabled and also check if there is modified `editor.trimAutoWhitespace` setting in user setting or workspace.

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! We figured it's missing some basic information or in some other way doesn't follow our [issue reporting guidelines](https://aka.ms/vscodeissuereporting). Please take the time to review these and update the issue.

Happy Coding!

**Comment by hegerdes** [comment]
@rebornix Thanks for your reply. I noticed that there was an update for VSCode formatting for notebooks, but it did not fix my issue. Thanks to your reply I also noticed that there are different trim settings: `editor.trimAutoWhitespace` and  `files.trimTrailingWhitespace`. Both are set to `true` on my side.  All extensions disabled.

What does not seem to work is the `trimTrailingWhitespace` part. Whitespaces on the end of the line and empty lines with whitespaces are not cleared. I created a short video for demonstration. See line 7,9 and 13.

https://github.com/microsoft/vscode/assets/40325561/0454fd6d-7a58-4ae9-86ff-435e5d1b4bd9

I hope that helps differentiating the problem
